src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -2885,6 +2885,289 @@ public static void IndexOfAny_InvalidCount_ThrowsArgumentOutOfRangeException(int
             AssertExtensions.Throws<ArgumentOutOfRangeException>("count", () => "foo".IndexOfAny(new char[] { 'o' }, startIndex, count));
         }
 
+        [Fact]
+        public static void ZeroLengthIndexOf_Char()
+        {
+            string empty = string.Empty;
+            int idx = empty.IndexOf((char)0);
+            Assert.Equal(-1, idx);
+
+            ReadOnlySpan<char> sp = new ReadOnlySpan<char>(Array.Empty<char>());
+            idx = sp.IndexOf((char)0);
+            Assert.Equal(-1, idx);
+        }
+
+        [Fact]
+        public static void IndexOfSequenceMatchAtStart_Char()
+        {
+            string s1 = "5172377457778667789";
+            string s2 = "517";
+            int index = s1.IndexOf(s2);
+            Assert.Equal(0, index);
+            Assert.Equal(index, s1.IndexOf(s2, StringComparison.Ordinal));
+
+            ReadOnlySpan<char> span = new ReadOnlySpan<char>(new char[] { '5', '1', '7', '2', '3', '7', '7', '4', '5', '7', '7', '7', '8', '6', '6', '7', '7', '8', '9' });
+            ReadOnlySpan<char> value = new ReadOnlySpan<char>(new char[] { '5', '1', '7' });
+            index = span.IndexOf(value);
+            Assert.Equal(0, index);
+            Assert.Equal(index, span.IndexOf(value, StringComparison.Ordinal));
+        }
+
+        [Fact]
+        public static void IndexOfSequenceMultipleMatch_Char()
+        {
+            string s1 = "123123123";
+            string s2 = "23";
+
+            int index = s1.IndexOf(s2);
+            Assert.Equal(1, index);
+            Assert.Equal(index, s1.IndexOf(s2, StringComparison.Ordinal));
+
+            ReadOnlySpan<char> span = new ReadOnlySpan<char>(new char[] { '1', '2', '3', '1', '2', '3', '1', '2', '3' });
```

Comment:
Same here, elsewhere. No need to call the ctor.

Reply:
Better yet, let's just use the original strings, wherever possible.
`ReadOnlySpan<char> span = s1.AsSpan();`

This way we avoid the code duplication and possibility mismatch between the string and the char array.